src/rpc/common/Validators.h

```diff
@@ -107,6 +109,66 @@ struct Required final
     verify(boost::json::value const& value, std::string_view key) const;
 };
 
+/**
+ * @brief A validator that forbids a field to be present
+ * If there is a value provided, it will forbid the field only when the value equals
+ * If there is no value provided, it will forbid the field when the field shows up
+ */
+template <typename... T>
+class NotSupported;
+
+/**
+ * @brief A specialized NotSupported validator that forbids a field to be present when the value equals the given value
+ */
+template <typename T>
+class NotSupported<T> final
+{
+    T val;
+
+public:
+    [[nodiscard]] MaybeError
+    verify(boost::json::value const& value, std::string_view key) const
+    {
+        if (value.is_object() and value.as_object().contains(key.data()))
+        {
+            using boost::json::value_to;
+            auto const res = value_to<T>(value.as_object().at(key.data()));
+            if (val == res)
+                return Error{RPC::Status{
+                    RPC::RippledError::rpcNOT_SUPPORTED,
+                    fmt::format("Not supported field '{}'s value '{}'", std::string{key}, res)}};
+        }
+        return {};
+    }
+
+    NotSupported(T t)
+    {
+        val = t;
```

Comment:
probably should be in initialization list instead? `: value_{value}` or similar.